pkg/api/frontend_logging_test.go

```diff
@@ -0,0 +1,143 @@
+package api
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/getsentry/sentry-go"
+	glog "github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	log "github.com/inconshreveable/log15"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+type logScenarioFunc func(c *scenarioContext, logs []*log.Record)
+
+func logSentryEventScenario(desc string, event FrontendSentryEvent, fn logScenarioFunc) {
+	Convey(desc, func() {
+		logs := []*log.Record{}
+		frontendLogger.SetHandler(log.FuncHandler(func(r *log.Record) error {
+			logs = append(logs, r)
+			return nil
+		}))
+
+		sc := setupScenarioContext("/log")
+		hs := HTTPServer{}
+
+		sc.defaultHandler = Wrap(func(w http.ResponseWriter, c *models.ReqContext) Response {
+			sc.context = c
+			return hs.LogFrontendMessage(c, event)
+		})
+
+		sc.m.Post(sc.url, sc.defaultHandler)
+		sc.fakeReqWithParams("POST", sc.url, map[string]string{}).exec()
+		fn(sc, logs)
+	})
+}
+
+func TestFrontendLoggingEndpoint(t *testing.T) {
+	ts, _ := time.Parse("2006-01-02T15:04:05.000Z", "2020-10-22T06:29:29.078Z")
+
+	Convey("FrontendLoggingEndpoint", t, func() {
```

Comment:
Please rewrite without Convey, since we're converting to the standard library for tests.